src/components/VideoPlayerContexts/PlaybackContext.tsx

```diff
@@ -73,21 +76,25 @@ function PlaybackContextProvider({children}: ChildrenProps) {
     );
 
     const resetVideoPlayerData = useCallback(() => {
-        videoResumeTryNumberRef.current = 0;
-        stopVideo();
-        setCurrentlyPlayingURL(null);
-        setSharedElement(null);
-        setOriginalParent(null);
-        currentVideoPlayerRef.current = null;
-        unloadVideo();
+        // Play video is an async operation and if we call stop video before the promise is completed,
+        // it will throw a console error. So, we'll wait until the promise is resolved before stopping the video.
+        (playVideoPromiseRef.current ?? Promise.resolve()).then(stopVideo).finally(() => {
+            videoResumeTryNumberRef.current = 0;
+            setCurrentlyPlayingURL(null);
+            setSharedElement(null);
+            setOriginalParent(null);
+            currentVideoPlayerRef.current = null;
+            unloadVideo();
+        });
     }, [stopVideo, unloadVideo]);
 
     useEffect(() => {
-        if (!currentReportID) {
+        // currentReportID is initially empty/-1 and we don't want to reset the video player data when it was previously empty/-1
```

Comment:
This comment says "what", but not "why". Can we explain why this is the case?

Reply:
Updated comment

Reply:
Great!